src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferStreamReadAsyncMemoryOverloadsTest : PreferStreamAsyncMemoryOverloadsTestBase
+    {
+        #region No diagnostic - Analyzer
+
+        [Fact]
+        public async Task NoDiagnosticAnalyzer_Read()
+        {
+            await AnalyzeAsync(@"
```

Comment:
Shall we have tests for vbnet too?

Reply:
Yes, but I want to focus on C# in this PR. I will address VB in a new PR.

Reply:
I don't think we have any reason to do a PR for half the functionality.  The basic VB tests should be added before this gets merged (unless there's some sort of date driven release I'm forgetting).

Reply:
Ok

Reply:
I added VB tests with the same logic as the C# and they are passing :)